stiebeleltron/integrationpluginstiebeleltron.cpp

```diff
@@ -87,36 +90,192 @@ void IntegrationPluginStiebelEltron::discoverThings(ThingDiscoveryInfo *info)
+                        thing->setStateValue(stiebelEltronOperatingModeStateTypeId, "Standby");
+                        break;
+                    case StiebelEltronModbusConnection::OperatingModeProgram:
+                        thing->setStateValue(stiebelEltronOperatingModeStateTypeId, "Program");
+                        break;
+                    case StiebelEltronModbusConnection::OperatingModeComfort:
+                        thing->setStateValue(stiebelEltronOperatingModeStateTypeId, "Comfort");
+                        break;
+                    case StiebelEltronModbusConnection::OperatingModeEco:
+                        thing->setStateValue(stiebelEltronOperatingModeStateTypeId, "Eco");
+                        break;
+                    case StiebelEltronModbusConnection::OperatingModeHotWater:
+                        thing->setStateValue(stiebelEltronOperatingModeStateTypeId, "Hot water");
+                        break;
+                }
+            });
+
+        connect(
+            connection, &StiebelEltronModbusConnection::systemStatusChanged, this,
+            [thing](uint16_t systemStatus) {
+                qCDebug(dcStiebelEltron()) << thing << "System status changed " << systemStatus;
+                thing->setStateValue(stiebelEltronPumpOneStateTypeId, systemStatus & (1 << 0));
+                thing->setStateValue(stiebelEltronPumpTwoStateTypeId, systemStatus & (1 << 1));
+                thing->setStateValue(stiebelEltronHeatingUpStateTypeId, systemStatus & (1 << 2));
+                thing->setStateValue(stiebelEltronAuxHeatingStateTypeId, systemStatus & (1 << 3));
+                thing->setStateValue(stiebelEltronHeatingStateTypeId, systemStatus & (1 << 4));
+                thing->setStateValue(stiebelEltronHotWaterStateTypeId, systemStatus & (1 << 5));
+                thing->setStateValue(stiebelEltronCompressorStateTypeId, systemStatus & (1 << 6));
+                thing->setStateValue(stiebelEltronSummerModeStateTypeId, systemStatus & (1 << 7));
+                thing->setStateValue(stiebelEltronCoolingModeStateTypeId, systemStatus & (1 << 8));
+                thing->setStateValue(stiebelEltronDefrostingStateTypeId, systemStatus & (1 << 9));
+                thing->setStateValue(stiebelEltronSilentModeStateTypeId, systemStatus & (1 << 10));
+                thing->setStateValue(stiebelEltronSilentMode2StateTypeId, systemStatus & (1 << 11));
+            });
+
+        connect(connection, &StiebelEltronModbusConnection::sgReadyStateChanged, this,
+                [thing](StiebelEltronModbusConnection::SmartGridState smartGridState) {
+                    qCDebug(dcStiebelEltron())
+                        << thing << "SG Ready activation changed" << smartGridState;
+                    switch (smartGridState) {
+                        case StiebelEltronModbusConnection::SmartGridStateModeOne:
+                            thing->setStateValue(stiebelEltronSgReadyModeStateTypeId, "Mode 1");
```

Comment:
Please note that the values for the SG ready state need to match the interface. 
This is actually a bug in the plugininfo compiler that you where able to build it like this.

https://preview.nymea.io/documentation/resources/interfaces#smartgridheatpump

Reply:
See https://github.com/nymea/nymea-plugins-modbus/commit/5f029352bd1812deb69b9f9d5e2edd956ad9c1dd